### PR TITLE
Add unit test coverage for consumer module

### PR DESCRIPTION
**Pull request by bcscc**
### Summary of Changes

- Add `ConsumerEditorUtilsTest` export function tests (JSON, CSV, TSV) covering `getTableContent`, null handling, quote escaping, whitespace normalization
- Create `KafkaRecordTest` with tests for `createFor` factory methods (success/failure paths), computed properties (`keyText`, `valueText`, `errorText`), and default value handling
- Create `ConsumerProducerFieldConfigTest` covering `getValueObj()` type conversion logic for all primitive types (STRING, JSON, LONG, INTEGER, DOUBLE, FLOAT, BASE64, NULL) and error cases
- Add edge case tests to `ConsumerFilterTest` for `DOES_NOT_CONTAINS` and `REGEX` with null record key/value, and empty headers scenarios
- Fix KafkaRecord.createFor failure path using keyFormat for both keyFormat and valueFormat instead of using the valueFormat parameter

### Optional: Click-testing instructions

N/A — unit tests only.

### Optional: Any additional details or context that should be provided?

- `KafkaConsumerClient` private methods (`calculatePartitions`, `partitionOffsetsForStartOffset`, `partitionOffsetsForStartDate`) were evaluated for testability but skipped — they're not reachable through the public API without mocking the full `KafkaConsumer` chain
- The JSON export null-handling bug discovered during testing is addressed in a separate follow-up PR

#### Tests

- [x] Added new
- [ ] Updated existing
- [ ] Deleted existing

### CODE REVIEW

File: test/io/confluent/intellijplugin/consumer/editor/ConsumerEditorUtilsTest.kt [hunk 1]
```diff
@@ -158,4 +167,145 @@ class ConsumerEditorUtilsTest {
...
+        fun `should return empty JSON array for empty table`() {
+            val model = createTableModel(listOf("Col1", "Col2"), emptyList())
+            val result = ConsumerEditorUtils.getTableContent(model, "json")
+            assertEquals("[]", result.trim())
+        }
+
+        @Test
+        fun `should produce valid JSON with column names as keys for single row`() {
+            val model = createTableModel(listOf("Name", "Age"), listOf(listOf("Alice", "30")))
+            val result = ConsumerEditorUtils.getTableContent(model, "json")
```
Comment (flippingbits) [inline test/io/confluent/intellijplugin/consumer/editor/ConsumerEditorUtilsTest.kt:215]: nit: Can we check that `result` holds valid JSON?

File: src/io/confluent/intellijplugin/consumer/editor/KafkaRecord.kt [hunk 2]
```diff
@@ -80,7 +80,7 @@ data class KafkaRecord(
                     valueSize = 0,
                     headers = emptyList(),
                     keyFormat = keyFormat ?: KafkaRegistryFormat.UNKNOWN,
-                    valueFormat = keyFormat ?: KafkaRegistryFormat.UNKNOWN
+                    valueFormat = valueFormat ?: KafkaRegistryFormat.UNKNOWN
```
Comment (flippingbits) [inline src/io/confluent/intellijplugin/consumer/editor/KafkaRecord.kt:83]: Nice catch! Could we cover this with a test that assigns different values to the `keyFormat` and `valueFormat`?

**Review by flippingbits — DISMISSED** [review]
LGTM!

SonarQube reported 50+ issues but most (if not all?) of them seem to be related to the naming of the test cases. I feel we can ignore it.

File: src/io/confluent/intellijplugin/consumer/editor/KafkaRecord.kt (hunk 2 above)
Comment (bcscc) [reply src/io/confluent/intellijplugin/consumer/editor/KafkaRecord.kt:83]: Already added [here](https://github.com/confluentinc/intellij/pull/373/changes#diff-5a104b54685db3f3506a9927dd00287d1e3f90b9d60d9a0382f0152c9b95b012R206-R215)!

File: test/io/confluent/intellijplugin/consumer/editor/ConsumerEditorUtilsTest.kt (hunk 1 above)
Comment (bcscc) [reply test/io/confluent/intellijplugin/consumer/editor/ConsumerEditorUtilsTest.kt:215]: Thanks, added!

**Review by flippingbits — APPROVED** [review]
LGTM 🎉
